core/src/main/java/ai/timefold/solver/core/impl/domain/valuerange/ValueRangeCache.java

```diff
@@ -116,4 +135,9 @@ public <Value_> ValueRangeCache<Value_> buildCache(Collection<Value_> collection
 
     }
 
+    private enum CacheType {
```

Comment:
`CacheType` and `Builder` can be merged. 
Both are enums, both mean the exact same thing.
This is just a needless obfuscation.

Alternatively, `cacheType` can become a boolean.